categories/categories/category.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * @package     API
+ * @subpackage  com_api
+ *
+ * @author      Techjoomla <extensions@techjoomla.com>
+ * @copyright   Copyright (C) 2009 - 2019 Techjoomla, Tekdi Technologies Pvt. Ltd. All rights reserved.
+ * @license     GNU GPLv2 <http://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
+ */
+
+// No direct access.
+defined('_JEXEC') or die('Restricted access');
+
+JLoader::register('CategoriesModelCategory', JPATH_ROOT . '/administrator/components/com_categories/models/category.php');
+
+/**
+ * Category API resource class
+ *
+ * @package  API
+ * @since    1.6.0
+ */
+class CategoriesApiResourceCategory extends ApiResource
+{
+	/**
+	 * Get categories
+	 *
+	 * @return  object  Category details wrapped inside standard api response wrapper
+	 */
+	public function get()
+	{
+		$this->plugin->setResponse($this->getCategory());
+	}
+
+	/**
+	 * Save category - create / update
+	 *
+	 * @return  object  Category details wrapped inside standard api response wrapper
+	 */
+	public function post()
+	{
+		// $this->plugin->setResponse($this->saveCategory());
+	}
+
+	/**
+	 * Get category details
+	 *
+	 * @return  array
+	 *
+	 * @since   1.6.0
+	 */
+	public function getCategory()
+	{
+		// 1. Important to include category table first
+		JTable::addIncludePath(JPATH_ROOT . '/administrator/components/com_categories/tables');
+
+		// 2. Then, get an instance of the generic articles model
+		$model = JModelLegacy::getInstance('Category', 'CategoriesModel', array('ignore_request' => true));
+
+		// Set application parameters in model
+		$app   = JFactory::getApplication();
+		$input = $app->input;
+
+		// Important to get from input directly and not from $input->get->get()
+		$id = $input->get('id', 0, 'int');
```

Comment:
Please support getting the category by either id or alias, so both `/categories/category/59` and `/categories/category/sports` should work

Reply:
@coolbung 

How do we differentiate between if an alias is set or an id is set?

Consider an example with a 
1. category having `id` as `2` 
2. and another category having `alias` as `sports-2`

Current `com_api` router only handles id, how will it work for alias?

in above two cases:

`$id = $input->get('id', 0, 'int');`  will return `2` as `$id`

Reply:
I can think of 2 options
1. if the id is numeric assume it to be a id, if not alias
2. Support a X-Identifier header. The value for this header can be id (default) or alias. This tells the plugin which record to update. This method is already implemented in the user edit plugin by @twsvaishali

Reply:
3. or can we add trigger to let extend com_api router, something like

```onApiRoute (&segments, &$vars)```

and let plugins set `vars` as needed

I tried this out and I got both id and alias to work